### PR TITLE
Skip instead of throwing on invalid DAV data migration

### PR DESCRIPTION
**Pull request by Pytal**
When encountering invalid data instead of failing we will skip it

### CODE REVIEW

**Review by tcitworld — APPROVED** [review]
LGTM

**Comment by come-nc** [comment]
This means partial import/partial export when there are errors in the data.
How is the user notified that the export is partial?
How is the user notified that the import is partial?

**Comment by Pytal** [comment]
> This means partial import/partial export when there are errors in the data. How is the user notified that the export is partial? How is the user notified that the import is partial?

When invoked via the CLI then the message is displayed though from the web UI there is no mechanism to communicate these messages

We could add a way for interface agnostic messages to be communicated or make a general `PartialDataException` that would only communicate that the data migration was partial due to some invalid data or maybe something else if there's a better way, any further suggestions @come-nc?

**Comment by Pytal** [comment]
> This means partial import/partial export when there are errors in the data. How is the user notified that the export is partial? How is the user notified that the import is partial?

When invoked via the CLI then the message is displayed though from the web UI there is no mechanism to communicate these messages

We could add a way for interface agnostic messages to be communicated or maybe something else if there's a better way, any further suggestions @come-nc?

**Review by skjnldsv — APPROVED** [review]

File: apps/dav/lib/UserMigration/CalendarMigrator.php [hunk 1]
```diff
@@ -182,6 +182,9 @@ function (ICalendar $calendar) use ($user, $output) {
 		)));
 	}
 
+	/**
+	 * @throws CalendarMigratorException
```
Comment (skjnldsv) [inline apps/dav/lib/UserMigration/CalendarMigrator.php:186]:
```suggestion
	 * @throws InvalidCalendarException
```

**Pull request merged**